protocol/protocol.go

```diff
@@ -90,8 +90,8 @@ func (p Protocol) ReadFunctionPublicKey() (*entities.FunctionPublicKey, error) {
 	return readMsg[entities.FunctionPublicKey](p.Websocket)
 }
 
-func (p Protocol) WriteFunctionPublicKey(key string) error {
-	return writeMsg(p.Websocket, entities.FunctionPublicKey{FunctionTokenPublicKey: key})
+func (p Protocol) WriteFunctionPublicKey(key string, name string) error {
```

Comment:
think it was mentioned in the related PR in enclave-supervisor but this should be renamed to something more generic.

Reply:
Changed the name!